sdk/core/Azure.Core.TestFramework/src/TestEnvironment.cs

```diff
@@ -214,12 +214,23 @@ public virtual TokenCredential Credential
                     }
                     else
                     {
+                        // If the recording is null but we are in Record Mode this means the Credential is being used
+                        // outside of a test. Attempt to use the env vars, but don't cache the credential so that subsequent
+                        // usages of this property that are within a test will store the variables in the recording.
+                        if (_recording == null)
```

Comment:
When would this typically happen (using record mode outside of a test)?

Reply:
It's used by the test framework itself to extend the resource group expiration.

Reply:
Ah right - let's add a comment to mention that magic scenario :)